Tests fail in tox, not with pytest outside of tox or on Travis-CI

When I try to run tox, there are four tests that fail in every environment:

```
$ tox -r -e py27
GLOB sdist-make: /Users/gback/code/cti-stix-validator/setup.py
py27 recreate: /Users/gback/code/cti-stix-validator/.tox/py27
py27 installdeps: pytest
py27 inst: /Users/gback/code/cti-stix-validator/.tox/dist/stix2-validator-0.0.1.zip
py27 installed: colorama==0.3.7,functools32==3.2.3.post2,jsonschema==2.5.1,py==1.4.32,pytest==3.0.5,requests==2.12.4,requests-cache==0.4.13,six==1.10.0,stix2-validator==0.0.1
py27 runtests: PYTHONHASHSEED='2876203125'
py27 runtests: commands[0] | py.test
========================================================= test session starts =========================================================
platform darwin -- Python 2.7.13, pytest-3.0.5, py-1.4.32, pluggy-0.4.0
rootdir: /Users/gback/code/cti-stix-validator, inifile: tox.ini
collected 134 items

lib/python2.7/site-packages/stix2validator/test/attack_pattern_tests.py ......
lib/python2.7/site-packages/stix2validator/test/bundle_tests.py ....
lib/python2.7/site-packages/stix2validator/test/custom_obj_tests.py .........
lib/python2.7/site-packages/stix2validator/test/identity_tests.py ....
lib/python2.7/site-packages/stix2validator/test/indicator_tests.py ................
lib/python2.7/site-packages/stix2validator/test/intrusion_set_tests.py ....
lib/python2.7/site-packages/stix2validator/test/malware_tests.py .....
lib/python2.7/site-packages/stix2validator/test/marking_definition_tests.py ......
lib/python2.7/site-packages/stix2validator/test/network_traffic_tests.py .F.F...
lib/python2.7/site-packages/stix2validator/test/observed_data_tests.py ...F..F..................................
lib/python2.7/site-packages/stix2validator/test/relationship_tests.py .........
lib/python2.7/site-packages/stix2validator/test/report_tests.py ..
lib/python2.7/site-packages/stix2validator/test/sighting_tests.py ....
lib/python2.7/site-packages/stix2validator/test/threat_actor_tests.py ......
lib/python2.7/site-packages/stix2validator/test/tool_tests.py .....
lib/python2.7/site-packages/stix2validator/test/vulnerability_tests.py ......

============================================================== FAILURES ===============================================================
__________________________________________ ObservedDataTestCases.test_network_traffic_ipfix ___________________________________________

self = <stix2validator.test.network_traffic_tests.ObservedDataTestCases testMethod=test_network_traffic_ipfix>

    def test_network_traffic_ipfix(self):
        net_traffic = copy.deepcopy(self.valid_net_traffic)
        net_traffic['objects']['1']['ipfix'] = {
            "minimumIpTotalLength": 32,
            "maximumIpTotalLength": 2556,
            "foo": "bar"
        }
>       self.assertFalseWithOptions(json.dumps(net_traffic))

lib/python2.7/site-packages/stix2validator/test/network_traffic_tests.py:95:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
lib/python2.7/site-packages/stix2validator/test/__init__.py:57: in assertFalseWithOptions
    self.assertEqual(results.is_valid, False)
E   AssertionError: True != False
________________________________________ ObservedDataTestCases.test_network_traffic_protocols _________________________________________

self = <stix2validator.test.network_traffic_tests.ObservedDataTestCases testMethod=test_network_traffic_protocols>

    def test_network_traffic_protocols(self):
        net_traffic = copy.deepcopy(self.valid_net_traffic)
        net_traffic['objects']['1']['protocols'].append('foobar')
>       self.assertFalseWithOptions(json.dumps(net_traffic))

lib/python2.7/site-packages/stix2validator/test/network_traffic_tests.py:82:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
lib/python2.7/site-packages/stix2validator/test/__init__.py:57: in assertFalseWithOptions
    self.assertEqual(results.is_valid, False)
E   AssertionError: True != False
_________________________________________ ObservedDataTestCases.test_directory_character_set __________________________________________

self = <stix2validator.test.observed_data_tests.ObservedDataTestCases testMethod=test_directory_character_set>

    def test_directory_character_set(self):
        observed_data = copy.deepcopy(self.valid_observed_data)
        observed_data['objects']['2'] = {
          "type": "directory",
          "path": "C:\\Windows\\System32",
          "path_enc": "blablabla"
        }
>       self.assertFalseWithOptions(json.dumps(observed_data))

lib/python2.7/site-packages/stix2validator/test/observed_data_tests.py:518:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
lib/python2.7/site-packages/stix2validator/test/__init__.py:57: in assertFalseWithOptions
    self.assertEqual(results.is_valid, False)
E   AssertionError: True != False
____________________________________________ ObservedDataTestCases.test_file_character_set ____________________________________________

self = <stix2validator.test.observed_data_tests.ObservedDataTestCases testMethod=test_file_character_set>

    def test_file_character_set(self):
        observed_data = copy.deepcopy(self.valid_observed_data)
        observed_data['objects']['0']['name_enc'] = "blablabla"
>       self.assertFalseWithOptions(json.dumps(observed_data))

lib/python2.7/site-packages/stix2validator/test/observed_data_tests.py:506:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
lib/python2.7/site-packages/stix2validator/test/__init__.py:57: in assertFalseWithOptions
    self.assertEqual(results.is_valid, False)
E   AssertionError: True != False
================================================ 4 failed, 130 passed in 9.13 seconds =================================================
ERROR: InvocationError: '/Users/gback/code/cti-stix-validator/.tox/py27/bin/py.test'
_______________________________________________________________ summary _______________________________________________________________
ERROR:   py27: commands failed
```


They all seem related to the code that gets data from the network and (optionally) caches it. I [tried to ensure the cache is never used during testing](https://github.com/oasis-open/cti-stix-validator/commit/2dace26814a9d1a6ebc5ce3fa8d993651a03517a), but that didn't seem to fix anything. 

The tests all pass when I run `pytest` outside of `tox`, and this doesn't seem to affect Travis-CI builds. @clenk , can you take a look at this, and see if the same thing is happening for you or if it's just my machine?

--- a/stix2validator/musts.py
+++ b/stix2validator/musts.py
@@ -5,6 +5,7 @@ import re
 from . import enums
 from .util import cyber_observable_check, has_cyber_observable_data
 from .errors import JSONError
+from .output import info
 
 
 def modified_created(instance):
@@ -211,6 +212,16 @@ def artifact_mime_type(instance):
                                     "Type of the form 'type/subtype'."
                                     % (key, obj['mime_type']), instance['id'])
 
+            else:
+                info("Can't reach IANA website; using regex for mime types.")
+                mime_pattern = '^(application|audio|font|image|message|model' \
+                               '|multipart|text|video)/[a-zA-Z0-9.+_-]+'
+                if not re.match(mime_pattern, obj['mime_type']):
+                    yield JSONError("The 'mime_type' property of object '%s' "
+                                    "('%s') should be an IANA MIME Type of the"
+                                    " form 'type/subtype'."
+                                    % (key, obj['mime_type']), instance['id'])
+
 
 @cyber_observable_check
 def character_set(instance):
@@ -225,10 +236,26 @@ def character_set(instance):
                                     "('%s') must be an IANA registered "
                                     "character set."
                                     % (key, obj['path_enc']), instance['id'])
+            else:
+                info("Can't reach IANA website; using regex for character_set.")
+                char_pattern = '^[a-zA-Z0-9_\(\)-]+$'
+                if not re.match(char_pattern, obj['path_enc']):
+                    yield JSONError("The 'path_enc' property of object '%s' "
+                                    "('%s') must be an IANA registered "
+                                    "character set."
+                                    % (key, obj['path_enc']), instance['id'])
 
         if ('type' in obj and obj['type'] == 'file' and 'name_enc' in obj):
             if enums.char_sets():
                 if obj['name_enc'] not in enums.char_sets():
+                    yield JSONError("The 'name_enc' property of object '%s' "
+                                    "('%s') must be an IANA registered "
+                                    "character set."
+                                    % (key, obj['name_enc']), instance['id'])
+            else:
+                info("Can't reach IANA website; using regex for character_set.")
+                char_pattern = '^[a-zA-Z0-9_\(\)-]+$'
+                if not re.match(char_pattern, obj['name_enc']):
                     yield JSONError("The 'name_enc' property of object '%s' "
                                     "('%s') must be an IANA registered "
                                     "character set."

--- a/stix2validator/shoulds.py
+++ b/stix2validator/shoulds.py
@@ -756,7 +756,7 @@ def protocols(instance):
                                         'protocols')
                 else:
                     info("Can't reach IANA website; using regex for protocols.")
-                    prot_pattern = '^[a-zA-Z0-9-]{1,15}'
+                    prot_pattern = '^[a-zA-Z0-9-]{1,15}$'
                     if not re.match(prot_pattern, prot):
                         yield JSONError("The 'protocols' property of object "
                                         "'%s' contains a value ('%s') not in "

--- a/stix2validator/test/network_traffic_tests.py
+++ b/stix2validator/test/network_traffic_tests.py
@@ -78,7 +78,7 @@ class ObservedDataTestCases(ValidatorTest):
 
     def test_network_traffic_protocols(self):
         net_traffic = copy.deepcopy(self.valid_net_traffic)
-        net_traffic['objects']['1']['protocols'].append('foobar')
+        net_traffic['objects']['1']['protocols'].append('foo_bar')
         self.assertFalseWithOptions(json.dumps(net_traffic))
         self.check_ignore(json.dumps(net_traffic), 'protocols')
 
@@ -90,7 +90,7 @@ class ObservedDataTestCases(ValidatorTest):
         net_traffic['objects']['1']['ipfix'] = {
             "minimumIpTotalLength": 32,
             "maximumIpTotalLength": 2556,
-            "foo": "bar"
+            "Foo": "bar"
         }
         self.assertFalseWithOptions(json.dumps(net_traffic))
         self.check_ignore(json.dumps(net_traffic), 'ipfix')

--- a/stix2validator/test/observed_data_tests.py
+++ b/stix2validator/test/observed_data_tests.py
@@ -502,7 +502,7 @@ class ObservedDataTestCases(ValidatorTest):
 
     def test_file_character_set(self):
         observed_data = copy.deepcopy(self.valid_observed_data)
-        observed_data['objects']['0']['name_enc'] = "blablabla"
+        observed_data['objects']['0']['name_enc'] = "bla.bla.bla"
         self.assertFalseWithOptions(json.dumps(observed_data))
 
         observed_data['objects']['0']['name_enc'] = "ISO-8859-2"
@@ -513,7 +513,7 @@ class ObservedDataTestCases(ValidatorTest):
         observed_data['objects']['2'] = {
           "type": "directory",
           "path": "C:\\Windows\\System32",
-          "path_enc": "blablabla"
+          "path_enc": "bla.bla.bla"
         }
         self.assertFalseWithOptions(json.dumps(observed_data))
 

